### PR TITLE
Add scripting to CubePilot CANMod

### PR DESCRIPTION
**Pull request by bugobliterator**
* Enables and fixes build for CubePilot CANMod with scripting support

### CODE REVIEW

**Review by magicrub — APPROVED** [review]
This fixes my Periph Scripting problem too!

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 1]
```diff
@@ -638,6 +638,7 @@ int AP_HAL__I2CDevice_read_registers(lua_State *L) {
     return success;
 }
 
+#if HAL_GCS_ENABLED
```
Comment (tridge) [inline libraries/AP_Scripting/lua_bindings.cpp:641]: we should allow UART ops without GCS enabled
see my alternative PR with a web gui here:
https://github.com/ArduPilot/ardupilot/pull/25970
that PR also fixes networking ops from scripting, and socket ops

**Review by tridge — CHANGES_REQUESTED** [review]
should merge the other PR first

File: libraries/AP_Scripting/lua_bindings.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/AP_Scripting/lua_bindings.cpp:641]: the other PR https://github.com/ArduPilot/ardupilot/pull/25970 is merged

Comment (bugobliterator) [reply libraries/AP_Scripting/lua_bindings.cpp]: Fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**